src/main/scala/org/allenai/wikitables/WikiTablesEntityLinker.scala

```diff
@@ -0,0 +1,207 @@
+package org.allenai.wikitables
+
+import scala.collection.JavaConverters._
+
+import java.util.Collection
+import edu.stanford.nlp.sempre.tables.`match`.EditDistanceFuzzyMatcher
+import edu.stanford.nlp.sempre.Formula
+import edu.stanford.nlp.sempre.FuzzyMatchFn.FuzzyMatchFnMode
+import scala.collection.mutable.ListBuffer
+import scala.collection.mutable.{Set => MutableSet}
+import org.allenai.pnp.semparse.Span
+import edu.stanford.nlp.sempre.tables.TableKnowledgeGraph
+import scala.util.Try
+import java.nio.file.Paths
+import java.nio.file.Files
+import com.google.common.base.Preconditions
+import scala.io.Source
+
+class WikiTablesEntityLinker {
+  
+  import WikiTablesEntityLinker._
+
+  def loadDataset(filename: String,
+      examples: Seq[WikiTablesExample]
+  ): Vector[RawEntityLinking] = {
+    val preprocessedFile = filename + PREPROCESSING_SUFFIX
+    
+    val entityLinkings = if (Files.exists(Paths.get(preprocessedFile))) {
+      RawEntityLinking.fromJsonFile(preprocessedFile)
+    } else {
+      val linkings = examples.map(getEntityLinking)
```

Comment:
I guess if the caching stuff you say is true, it probably isn't...  That's unfortunate.